tests/PactBuilder/JsonPatternBuilderTest.php

```diff
@@ -17,87 +18,231 @@ public function testItBuildsMessage(): void
     {
         $builder = new JsonPatternBuilder();
         $builder = $builder->withPattern([
-            'only' => 'value',
-            'type' => $builder->like('value'),
-            'deep_only' => $builder->eachLike([
-                'some' => 'value',
-            ]),
-            'deep_type' => $builder->eachLike([
-                'type' => $builder->like('value'),
-                'some' => $builder->term('value', '/.*/'),
-            ]),
-            'some' => $builder->term('value', '/.*/'),
-            'another' => [
-                'deep' => $builder->term('value', '/.*/'),
-            ],
-            'array' => [
-                $builder->term('value', '/.*/'),
-            ],
+            'some' => 'value',
         ]);
 
         $body = $builder->build();
 
         Assert::assertSame(
-            '{"only":"value","type":"value","deep_only":{"some":"value"},"deep_type":{"type":"value","some":"value"},"some":"value","another":{"deep":"value"},"array":["value"]}',
+            '{"some":"value"}',
             $body->content()
         );
 
-        Assert::assertSame([
-            '$.type' => [
-                'combine' => 'AND',
+        Assert::assertSame([], $body->matchingRules()->toArray());
+    }
+
+    public function testItIsImmutable(): void
+    {
+        $builder = new JsonPatternBuilder();
+
+        Assert::assertNotSame($builder, $builder->withPattern([]));
```

Comment:
Maybe do `assertEquals` as well to make sure the `assertSame` does not just fail because of a change in the object's state?